src/test/java/com/kodilla/ecommercee/entity/CartEntityTest.java

```diff
@@ -1,7 +1,147 @@
+
+        long userId = user.getId();
+
+
+        //then
+        Assert.assertTrue(cart.isPresent());
+        Assert.assertEquals(Optional.of(userId), Optional.of(cart.get().getUserEntity().getId()));
+        Assert.assertEquals(1, cart.get().getProductMap().size());
+
+        // cleanUp
+        cartRepository.deleteById(id);
+        userRepository.deleteById(userId);
+        long productId = pen.getId();
+        productRepository.deleteById(productId);
+    }
+
+    @Test
+    public void cardDeletingUserNotDeletingTest(){
+        //given
+        UserEntity user = new UserEntity();
+        Map<ProductEntity, Integer> products = new HashMap<>();
+        CartEntity newCart = new CartEntity(user, products);
+
+        //when
+        userRepository.save(user);
+        cartRepository.save(newCart);
+
+        //then
+        int id = newCart.getCartId();
+        long userId = user.getId();
+        cartRepository.deleteById(id);
+        Optional<UserEntity> newUser = userRepository.findById(userId);
+        Optional<CartEntity> cart = cartRepository.findById(id);
+        Assert.assertFalse(cart.isPresent());
+        Assert.assertTrue(newUser.isPresent());
+
+        // cleanUp
+        userRepository.deleteById(userId);
+    }
+
+    @Test
+    public void cardDeletingProductsNotDeletingTest(){
+        //given
+        UserEntity user = new UserEntity();
+        ProductEntity pen = new ProductEntity("pen", new BigDecimal(1.5), "description");
+        Map<ProductEntity, Integer> products = new HashMap<>();
+        products.put(pen, 4);
+        CartEntity newCart = new CartEntity(user, products);
+
+        //when
+        productRepository.save(pen);
+        userRepository.save(user);
+        cartRepository.save(newCart);
+
+
+        //then
+        int id = newCart.getCartId();
+        long penId = pen.getId();
+        Optional<ProductEntity> newPen = productRepository.findById(penId);
+        cartRepository.deleteById(id);
+        Optional<CartEntity> cart = cartRepository.findById(id);
+
+        Assert.assertFalse(cart.isPresent());
+        Assert.assertTrue(newPen.isPresent());
+
+        // cleanUp
+        long userId = user.getId();
+        userRepository.deleteById(userId);
+        productRepository.deleteById(penId);
+    }
+
+    @Test
+    public void addingProductToCartTest(){
+        //given
+        UserEntity user = new UserEntity();
+        ProductEntity pen = new ProductEntity("pen", new BigDecimal(1.5), "description");
+        Map<ProductEntity, Integer> products = new HashMap<>();
+        CartEntity newCart = new CartEntity(user, products);
+
+        //when
+        userRepository.save(user);
+        productRepository.save(pen);
+        cartRepository.save(newCart);
+
+        int cartId = newCart.getCartId();
+        Optional<CartEntity> cart = cartRepository.findById(cartId);
+        cart.get().getProductMap().put(pen, 3);
```

Comment:
Mam tylko pytanie czy dodanie produktu do mapy nie powinno być przed zapisem?

Reply:
wiesz co,  tak myślałam najpierw i tak jest we wcześniejszym teście, gdzie sprawdzam czy produkty się zapisały razem z cart,, ale chciałam tutaj przetestować, czy jesteśmy w stanie dodać produkt do już istniejącego zamówienia.

Reply:
cart.get().getProductMap().put(pen, 3);
Asser.... 
Jak zrobisz tak to nie wiem czy nie sprawdzi po prostu że obiekt cart zawiera to a nie ten znajdujący się w bazie danych. A jak tak mówisz że już chcesz do dodanego to może trzeba po tym jeszcze zapisać i dopiero wtedy pobrać i sprawdzic, czy to się samo aktualizuje w bazie danych

Reply:
?

Reply:
Kurczę. Czyli jeszcze raz zrobić zapisywanie? Tutaj w tym teście wrzucam pustą hashmapę na początku, a po tej operacji: cart.get().getProductMap().put(pen, 3); sprawdzam assercję i podaje, że w mapie jest jeden rekord, więc tak to wygląda, że się zapisał.

Reply:
i w assercji też podaje ten cart z bazy danych  Assert.assertEquals(1, cart.get().getProductMap().size());

Reply:
czy to jest referencja tylko do tego, a nie odwołanie do rekordu w DB?

Reply:
Pobiore twój branch i sprawdzę ale musisz zaczekać

Reply:
Wrzucona metoda zapisująca rekord w DB, dzięki :)